### PR TITLE
Implemented game server pinging

### PR DESCRIPTION
**Pull request by Laugslander**

### CODE REVIEW

**Comment by TehMessiah** [comment]
SonarQube analysis reported 4 issues
* MAJOR 2 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [GameServerForMainServer.java#L16](https://github.com/PTS3-S34A/Game-Server/blob/5699d7b7149d16fb128d772bd8097a1feb2ae7de/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerForMainServer.java#L16): Override this superclass' "equals" method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS2160)
1. MAJOR [GameServerForMainServer.java#L18](https://github.com/PTS3-S34A/Game-Server/blob/5699d7b7149d16fb128d772bd8097a1feb2ae7de/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerForMainServer.java#L18): Remove this unused "controller" private field. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1068)
1. MINOR [GameServerForMainServer.java#L20](https://github.com/PTS3-S34A/Game-Server/blob/5699d7b7149d16fb128d772bd8097a1feb2ae7de/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerForMainServer.java#L20): Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Review by BeeHiveJava — APPROVED** [review]

**Comment by BeeHiveJava** [comment]
Graag conflicts fixen.

File: Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerForMainServer.java [hunk 1]
```diff
@@ -27,8 +28,13 @@ public boolean createSession(String name, String password, String hostName, int
...
+    public long getAvailableMemory() throws RemoteException {
+        return Runtime.getRuntime().freeMemory();
+    }
+
+    @Override
+    public int ping(int value) throws RemoteException {
+        return value * RmiConstants.PING_CALCULATION_FACTOR;
     }
 
     public final void close() throws IOException {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerForMainServer.java:40]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Comment by ghost** [comment]
SonarQube analysis reported 4 issues
* MAJOR 2 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [GameServerForMainServer.java#L17](https://github.com/PTS3-S34A/Game-Server/blob/b559c6dabd2da7a08a5daeb2f4df33be57f11439/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerForMainServer.java#L17): Override this superclass' "equals" method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS2160)
1. MAJOR [GameServerForMainServer.java#L19](https://github.com/PTS3-S34A/Game-Server/blob/b559c6dabd2da7a08a5daeb2f4df33be57f11439/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerForMainServer.java#L19): Remove this unused "controller" private field. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1068)
1. MINOR [GameServerForMainServer.java#L21](https://github.com/PTS3-S34A/Game-Server/blob/b559c6dabd2da7a08a5daeb2f4df33be57f11439/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerForMainServer.java#L21): Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Pull request merged**